### PR TITLE
Podcast rss

### PR DESCRIPTION
**Pull request by joshr4**

### CODE REVIEW

File: db/migrations/20231009165641_add_ext_rss_feeds.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+exports.up = function (knex) {
+  return knex.schema.createTable("feeds", function (table) {
+    table.increments("id").primary().unique();
+    table.string("feed_url").notNullable();
```
Comment (blastshielddown) [inline db/migrations/20231009165641_add_ext_rss_feeds.js:4]:
```suggestion
    table.text("feed_url").notNullable();
```

File: controllers/rssFeeds.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  language: string;
+  rel: string;
+  type: string;
+  url: string;
+};
+type Value = {
+  method: string;
+  suggested: string;
+  type: string;
+  recipients: any[];
```
Comment (blastshielddown) [inline controllers/rssFeeds.ts:62]: We may need to add custom key/values to this type to account for any in the value blocks (Wavlake uses custom key/values, for example)

File: db/migrations/20231009165641_add_ext_rss_feeds.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+exports.up = function (knex) {
+  return knex.schema.createTable("external_feed", function (table) {
+    table.increments("id").primary().unique();
```
Comment (blastshielddown) [inline db/migrations/20231009165641_add_ext_rss_feeds.js:3]: Should we use the podcast `guid` here instead of a generic `id`?

Comment (joshr4) [reply db/migrations/20231009165641_add_ext_rss_feeds.js:3]: i wasnt having much luck finding a podcast `guid` out in the wild yet, doesnt look like its present for the two sample podcasts feeds i was using, and i didnt spot it on podcastindex, but i haven't spent a ton of time exploring podcastindex yet

Comment (joshr4) [reply db/migrations/20231009165641_add_ext_rss_feeds.js:3]: just double checked a raw xml feed and i do see one in boostagram ball, ill have to check the parser lib's code to see where the disconnect is in grabbing that attribute

Comment (joshr4) [reply db/migrations/20231009165641_add_ext_rss_feeds.js:3]: added the `podcast:guid` key 
19f8b1e1204532a312810db76fb29bd99db18425
and aliased it to `guid`
df97ed46d103af14abd8ba92f4efe55bc4d0e555

File: db/migrations/20231012154828_add_ext_rss_feeds.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+exports.up = function (knex) {
+  return knex.schema.createTable("external_feed", function (table) {
+    table.uuid("guid").primary().unique();
+    table.text("feed_url").notNullable();
```
Comment (blastshielddown) [inline db/migrations/20231012154828_add_ext_rss_feeds.js:4]:
```suggestion
    table.text("title").notNullable();
```

We can probably drop this now right? Or we could replace with a human readable `title` so it's easy to tell what the record is for

**Review by blastshielddown — COMMENTED** [review]
We'll have to add references to the new env vars to the `staging.cloudbuild.yaml` and `prod.cloudbuild.yaml` files to reference the podcastindex API keys

**Review by blastshielddown — APPROVED** [review]

**Pull request merged**